public/typescript/registration.ts

```diff
@@ -30,8 +30,9 @@ declare const publicKeyOptions: SerializedPublicKeyCredentialCreationOptions;
     updateState(S.REQUEST_USER_FOR_ATTESTATION),
     requestUserAttestation,
     updateState(S.PUBLIC_KEY_CREDENTIALS),
-    concatMap((credentials) => isPublicKeyCredentialType(credentials) ?
-      of(credentials).pipe(
+    concatIfElse(
```

Comment:
Consider offloading the logic to some helper functions with more descriptive method names. This to lower the cyclomatic complexity.